sunpy/physics/transforms/differential_rotation.py

```diff
@@ -85,9 +81,335 @@ def diff_rot(ddays, latitude, rot_type='howard', frame_time='sidereal'):
 
         #This is in micro-radians / sec
         rotation_rate = A + B * sin2l + C * sin4l
-        rotation_deg = rotation_rate * 1e-6  * delta_seconds / np.deg2rad(1)
+        rotation_deg = rotation_rate * 1e-6 * delta_seconds / np.deg2rad(1)
 
     if frame_time == 'synodic':
         rotation_deg -= 0.9856 * delta_days
 
-    return Longitude((np.round(rotation_deg.value,4)),u.deg)
+    #return Longitude((np.round(rotation_deg, 4)), u.deg)
+    return np.round(rotation_deg, 4) * u.deg
+
+
+@quantity_input(x=u.arcsec, y=u.arcsec)
+def rot_hpc(x, y, tstart, tend, spacecraft=None, frame_time='synodic', rot_type='howard', **kwargs):
+    """Given a location on the Sun referred to using the Helioprojective
+    Cartesian co-ordinate system (typically quoted in the units of arcseconds)
+    use the solar rotation profile to find that location at some later or
+    earlier time.
+
+    Parameters
+    -----------
+    x : `~astropy.units.Quantity`
+        helio-projective x-co-ordinate in arcseconds (can be an array)
+    y : `~astropy.units.Quantity`
+        helio-projective y-co-ordinate in arcseconds (can be an array)
+
+    tstart : `sunpy.time.time`
+        date/time to which x and y are referred.
+    tend : `sunpy.time.time`
+        date/time at which x and y will be rotated to.
+    spacecraft : { None | "soho" | "stereo_a" | "stereo_b" }
```

Comment:
Drop what?  The spacecraft keyword argument?

On Thu, Feb 5, 2015 at 8:50 AM, Stuart Mumford <notifications@github.com>
wrote:

> In sunpy/physics/transforms/differential_rotation.py
> <https://github.com/sunpy/sunpy/pull/1281#discussion_r24163677>:
>
> > +    Cartesian co-ordinate system (typically quoted in the units of arcseconds)
> > +    use the solar rotation profile to find that location at some later or
> > +    earlier time.
> > +
> > +    Parameters
> > +    -----------
> > +    x : `~astropy.units.Quantity`
> > +        helio-projective x-co-ordinate in arcseconds (can be an array)
> > +    y : `~astropy.units.Quantity`
> > +        helio-projective y-co-ordinate in arcseconds (can be an array)
> > +
> > +    tstart : `sunpy.time.time`
> > +        date/time to which x and y are referred.
> > +    tend : `sunpy.time.time`
> > +        date/time at which x and y will be rotated to.
> > +    spacecraft : { None | "soho" | "stereo_a" | "stereo_b" }
>
> We should drop this. It's not supported at the moment and will not need to
> be explicitly specified when using the proper coordinates, as the
> coordinate framework handles the location of the observer when doing the
> transforms.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/sunpy/sunpy/pull/1281/files#r24163677>.
>